components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1596,11 +1596,14 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
                 { teamId, chargebeeSubscriptionId },
             );
         }
-        const subscriptionId = await this.stripeService.findUncancelledSubscriptionByAttributionId(
-            AttributionId.render({ kind: "team", teamId: teamId }),
-        );
-        if (subscriptionId) {
-            await this.stripeService.cancelSubscription(subscriptionId);
+        const billingMode = await this.billingModes.getBillingMode({ kind: "team", teamId: teamId }, now);
+        if (billingMode.mode === "usage-based") {
```

Comment:
Just out of curiosity: wouldn't it be simpler (and safer) to check for "isPaymentEnabled" instead of invoking the complicated billingMode logic?

Reply:
I can't find any reference to `isPaymentEnabled`. Do you have a link to what you mean?

Reply:
Sure, sorry for the inaccuracy. It's just a simple config flag, which seems a bit more straightforward than the complicated state-dependent logic of billingModes:

https://github.com/gitpod-io/gitpod/blob/2f0b9462b0451f0974434fa36bd757420ef04e34/components/server/ee/src/user/user-deletion-service.ts#L28

Reply:
Alternatively (but it's pretty much the same thing), we could also check whether Stripe is configured, because that's the requirement here:

https://github.com/gitpod-io/gitpod/blob/17d7063b42a419a7e0f14766ad393a2394dbb211/components/server/ee/src/user/stripe-service.ts#L24

I'd personally be happy with any of these two approaches 😊

Reply:
Thanks 🙏 . I think you're right - a simple `enablePayment` check suffices here.

PR rebased to do that.